pkg/labels/arraylist.go

```diff
@@ -142,3 +184,86 @@ func (lsp *LabelArrayList) MergeSorted(include LabelArrayList) LabelArrayList {
 	*lsp = merged
 	return *lsp
 }
+
+func nextArray(str string, end int) (int, int) {
+	start := strings.IndexByte(str[end:], '[')
+	if start >= 0 {
+		start += end
+		end = strings.IndexByte(str[start:], ']')
+		if end >= 0 {
+			end += start + 1
+		}
+	}
+	return start, end
+}
+
+func writeRemainder(str string, start, end int, sb *strings.Builder) {
+	if start >= 0 && start < end {
+		if sb.Len() > 0 {
+			sb.WriteString(", ")
+		}
+		sb.WriteString(str[start:])
+	}
+}
+
+// merge 'b' to 'a' assuming both are sorted
+func MergeSortedLabelArrayListStrings(a, b string) string {
```

Comment:
I wonder if this could benefit from some unit tests. Or is the logic covered by existing tests indirectly?

Reply:
Will add unit tests, as it looks like current test do not cover this.